#96 graphics

anthonyl6 opened the pull request:
- Added FPS counter and limiter.
- Separated GUI elements and scripts

anthonyl6 committed:
Resolved most of the comments but new bugs came

anthonyl6 committed:
Fixed advanced options

anthonyl6 committed:
Added a fps limiter and a dropdown to select fps presets

anthonyl6 committed:
Added a fps counter and finished fps limiter

anthonyl6 committed:
Fixed quality switching bug

anthonyl6 committed:
Finished fps counter but graphics don't update on load until the menu opens

anthonyl6 committed:
Merge remote-tracking branch 'origin/development' into #96-graphics

anthonyl6 committed:
Revamped objects testing

anthonyl6 committed:
Merge remote-tracking branch 'origin/development' into test

anthonyl6 committed:
Graphics load on start

anthonyl6 committed:
Merge remote-tracking branch 'origin/development' into #96-graphics

anthonyl6 committed:
Inspector objects are gone

anthonyl6 committed:
Update

anthonyl6 committed:
Merge

anthonyl6 committed:
Updated small things

anthonyl6 committed:
Fixed loading player pref bug

anthonyl6 committed:
Changes version number

Lithrun commented on Assets/Scripts/UI/InGame/Graphics/GeneralGraphics.cs:
```diff
@@ -52,88 +53,50 @@ public Toggle CustomSettings
+		
+		
+
+		public enum Textures
```
Public enum should be a separate file, or make this private if it's only used within this class

Lithrun commented on Assets/Scripts/UI/InGame/Graphics/GraphicsActions.cs:
```diff
@@ -0,0 +1,10 @@
+﻿using System.Collections;
```
Empty class, so we can delete this

Lithrun commented on Assets/Scripts/UI/InGame/Graphics/GraphicsController.cs:
```diff
@@ -57,45 +38,72 @@ public void SaveGraphicPlayerPrefs()
+			}
+
+			GeneralGraphic.UpdateEverything();
+			//QualitySwitcher.UpdateQuality();
```
Comment

Lithrun commented on Assets/Scripts/UI/InGame/Graphics/GraphicsController.cs:
```diff
@@ -57,45 +38,72 @@ public void SaveGraphicPlayerPrefs()
 			if(!selected)
 			{
-				QualitySettings.SetQualityLevel(temp);
+				GeneralGraphic.TextureQuality.interactable = false;
```
These six `GeneralGraphic.TextureQuality.interactable = false` can also just be replaced with `GeneralGraphic.TextureQuality.interactable = selected`, so if we aren't selected, we set them to false, otherwise they become true.

Lithrun commented on Assets/Scripts/UI/InGame/Graphics/GraphicsController.cs:
```diff
@@ -57,45 +38,72 @@ public void SaveGraphicPlayerPrefs()
+				QualitySwitcher.UpdateQuality();
+				
 			}
 			if(selected)
```
Should be an else statement

Lithrun commented on Assets/Scripts/UI/InGame/Graphics/GraphicsView.cs:
```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections;
```
Empty class

Lithrun commented on Assets/Scripts/UI/InGame/Graphics/ResolutionSwitcher.cs:
```diff
@@ -72,11 +65,11 @@ public void LoadPlayerPrefs()
 			DropDownLabel.text = resolution;
 			if(screen_mode == 1)
```
`if(screen_mode == 1) Toggle.isOn = true else Toggle.isOn = false` can be made shorter
`Toggle.isOn = screen_mode == 1`

Lithrun commented on Assets/Scripts/UI/InGame/HUD/FPSCounter.cs:
```diff
@@ -0,0 +1,35 @@
+	}
+
+	private void Start() {
+		time = 0f;
```
0f is the default value of an unassigned float, so no need to assign the default here

Lithrun reviewed (changes requested):
Besides the 8 comments
- [x] A string named "saved player prefs" remains on the screen after closing the Graphic Settings Gui
- [x] There's a typo in Medium, it's spelled as "Meduim" atm
- [x] Selecting profiles such as "Fastest" is not actually the fastest, as texture quality is on medium instead of low. With fastest the lowest graphical settings are expected. The same applies for "Fantastic" quality which has shadows disabled and textures set to medium, whereas it's expected that the best quality is there
- [x] There's a slider under FPS dropdown which doesn't seem to do anything
- [x] Why a slider instead of an input field for FPS? I'd like to set my FPS to 144 and my options are either 120 and 240 now. Remove that limitation.
- [x] FPS limit is not persisted. Setting it to 240, hitting save and upon starting it is not present
- [x] Upon launching the game there's something called "Graphics Quality". Changing this value does nothing which can be confusing to players.
- [x] The FPS meter is overlapping with the builder number.
- [x] StressTest build had an outdated version of 96, but please confirm that #142 #143 have already been fixed

anthonyl6 committed:
Fixed all the suggestions.

anthonyl6 committed:
fixed typo

anthonyl6 committed:
Set defaults and load player prefs on startup.

anthonyl6 committed:
Bugs seem to  be different in the editor than in a build??

anthonyl6 committed:
Fixed issues 142 and 143 as well as fps counter not moving with resolution change

anthonyl6 committed:
Fixed quick bug

anthonyl6 committed:
update

anthonyl6 committed:
Merged with development

anthonyl6 committed:
Updated unity version

anthonyl6 committed:
Removed line of code that broke graphics scripts

anthonyl6 committed:
Removed the update function from general graphics and called it only when necessary.

anthonyl6 committed:
Anchored the fps counter to top left

anthonyl6 committed:
Turned off HUD object visibility

Lithrun commented on Assets/Scripts/Characters/Human/CustomCharacterManager.cs:
```diff
@@ -402,7 +402,8 @@ private void setStatPoint(CreateStat type, int pt)
     {
         int num;
-        QualitySettings.SetQualityLevel(5, true);
+        //IS THIS LINE OF CODE IMPORTANT??? IT AFFECTS GRAPHICS SCRIPTS
```
When players open the character customization page in AoTTG, it displays them at the highest quality.

anthonyl6 replied:
It doesn't happen when the player opens the character customization page it happens when they spawn in.

Lithrun commented on Assets/Scripts/FengGameManagerMKII.cs:
```diff
@@ -2144,7 +2144,8 @@ private void loadconfig()
             QualitySettings.vSyncCount = 1;
         }
         AudioListener.volume = PlayerPrefs.GetFloat("vol", 1f);
-        QualitySettings.masterTextureLimit = PlayerPrefs.GetInt("skinQ", 0);
```
This was how settings were previously stored and loaded upon opening the game. Since you have implemented a new way of doing it, this line is no longer required.

Lithrun replied:
If you marked this as resolved, then also remove the two lines of code
`      //IS THIS LINE OF CODE IMPORTANT?? IT AFFECTS GRAPHICS SCRIPTS!!
        //QualitySettings.masterTextureLimit = PlayerPrefs.GetInt("skinQ", 0);`

Lithrun commented on Assets/Scripts/UI/InGame/Graphics/FPSLimiter.cs:
```diff
@@ -0,0 +1,86 @@
+	[SerializeField] public Dropdown fpsDropdown;
+	
+
+	public Dropdown FPSDropdown
```
We should not have a dropdown for FPS limits.

Lithrun commented on Assets/Scripts/UI/InGame/Graphics/GraphicsController.cs:
```diff
@@ -11,91 +9,166 @@ public class GraphicsController : MonoBehaviour {
+				label.color = Color.green;
+				label.text = "saved player prefs";
+			}
+			catch(Exception ex)
```
Does it throw an exception here? If so, what is the expected exception? Using catch Exception should be avoided as this will not be logged.

Lithrun commented on Assets/Scripts/UI/InGame/Graphics/GraphicsController.cs:
```diff
@@ -11,91 +9,166 @@ public class GraphicsController : MonoBehaviour {
+			
 		}
+
+		// NULL REFERENCE HAPPENES HERE
```
If a null reference happens, then why isn't this fixed? Sounds like something that we need to investigate

anthonyl6 replied:
It was fixed just forgot to remove it\

Lithrun commented on Assets/Scripts/UI/InGame/Graphics/FPSLimiter.cs:
```diff
@@ -0,0 +1,86 @@
+			{
+				try
+				{
+					Application.targetFrameRate = int.Parse(text.ToString());
```
Instead of using Parse and expecting an exception, please use int.TryParse https://docs.microsoft.com/en-us/dotnet/api/system.int32.tryparse?view=netframework-4.6

Lithrun commented on Assets/Scripts/UI/InGame/Graphics/GraphicsController.cs:
```diff
@@ -11,91 +9,166 @@ public class GraphicsController : MonoBehaviour {
+			if(PlayerPrefs.HasKey("GraphicsData") && PlayerPrefs.HasKey("QualityProfile") && PlayerPrefs.HasKey("FPSLimit"))
+			{
+				LoadGraphicPlayerPrefs();
+				Debug.Log("Has Prefs");
```
Debug.Log leftover

Lithrun commented on Assets/Scripts/UI/InGame/Graphics/GraphicsController.cs:
```diff
@@ -11,91 +9,166 @@ public class GraphicsController : MonoBehaviour {
+			GeneralGraphic.UpdateEverything();
+		}
+
+		public void DeletePrefs()
```
"GraphicsData", "QualityProfile", "FPSLimit" are repeated multiple times throughout the classes. A cleaner approach would be to make these a const.

Lithrun reviewed (changes requested):
See notes. Additionally
- [x] Again the FPS meter is obstructing the view of other UI elements. Make it smaller and locate it somewhere where it does not overlap with existing UI. Currently playerlist will overlap if there are +5 players.
- [x] Default framelimit appears to be hardcoded on 30FPS. Make the default the refresh rate of the monitor https://docs.unity3d.com/ScriptReference/Resolution-refreshRate.html

anthonyl6 committed:
Fixed PR changes

anthonyl6 committed:
Removed fps limit

anthonyl6 committed:
Update

anthonyl6 committed:
Merge

anthonyl6 committed:
Removed the aot configuration on startup

anthonyl6 committed:
Re merged

Lithrun commented on Assets/Scripts/Gamemode/CaptureGamemode.cs:
```diff
@@ -103,6 +103,11 @@ public void AddTitanScore(int score)
             CheckWinConditions();
         }
 
+        private void awake()
```
Please remove this

anthonyl6 replied:
I think I accidentally applied stashed changes for issue #175 as I was messing around with it. This isnt supposed to be in there

anthonyl6 committed:
Set InGameUI object to false

anthonyl6 committed:
Merge into origin/development

anthonyl6 committed:
Removed debug

anthonyl6 committed:
Added display switch

anthonyl6 committed:
Revert "Added display switch"

This reverts commit 6f15b174965f0dc33a8e8865a487d02b67657cab.

anthonyl6 committed:
Merge dev

anthonyl6 committed:
Merge dev

anthonyl6 committed:
update version manager

anthonyl6 committed:
parent de23caaa732a7f04553f181c8daf8740c5f6b806
author ThatProBro2535 <agloria2001@gmail.com> 1592340543 -0600
committer ThatProBro2535 <agloria2001@gmail.com> 1592342259 -0600

Conflict

conflicts

Remove old code modifying vSync

Removed copy of code

Conflicts

mwnDK1402 committed:
Fix broken references

mwnDK1402 committed:
Remove old code modifying vSync

anthonyl6 committed:
Merging

anthonyl6 committed:
Removed testing code

anthonyl6 committed:
Merged dev

anthonyl6 committed:
Removed reference to wrong script

anthonyl6 committed:
Merge Dev
